src/SonarScanner.MSBuild.PreProcessor/AnalysisConfigGenerator.cs

```diff
@@ -94,50 +103,66 @@ public static AnalysisConfig GenerateFile(ProcessedArgs localSettings,
-            "sonar.cs.opencover.reportsPaths",
-        ];
-        var allProperties = localSettings.CmdLineProperties.GetAllProperties().ToList();
-        var coveragePaths = string.Join(",", allProperties.Where(x => coveragePropertyNames.Contains(x.Id)).Select(x => x.Value));
-        if (!localSettings.ScanAllAnalysis      // if scanAll analysis is disabled, we will not pick up the coverage files anyways
-            || coveragePaths.Length == 0)       // if there are no coverage files, there is nothing to exclude
+        var commandLineProperties = localSettings.CmdLineProperties.GetAllProperties().ToList();
+        var localCoveragePaths = string.Join(",", commandLineProperties.Where(x => CoveragePropertyNames.Contains(x.Id)).Select(x => x.Value));
+        var serverCoveragePaths = string.Join(",", serverProperties.Where(x => CoveragePropertyNames.Contains(x.Key)).Select(x => x.Value));
+        if (!localSettings.ScanAllAnalysis                                  // If scanAll analysis is disabled, we will not pick up the coverage files anyways
+            || localCoveragePaths.Length + serverCoveragePaths.Length == 0) // If there are no coverage files, there is nothing to exclude
         {
-            return allProperties;
+            return;
         }
+        var localExclusions = localSettings.GetSetting(SonarExclusions, string.Empty);
+        var serverExclusions = serverProperties.ContainsKey(SonarExclusions) ? serverProperties[SonarExclusions] : string.Empty;
+        AddCoverageExclusions(localCoveragePaths.Length > 0 ? localCoveragePaths : serverCoveragePaths);
 
-        if (allProperties.Find(x => x.Id == "sonar.exclusions") is { } exclusionsProperty)
+        void AddCoverageExclusions(string coveragePaths)
         {
-            exclusionsProperty.Value = exclusionsProperty.Value + "," + coveragePaths;
+            if (string.IsNullOrEmpty(localExclusions) && string.IsNullOrEmpty(serverExclusions))
+            {
+                localExclusions = coveragePaths;
+            }
+            else if (string.IsNullOrEmpty(localExclusions))
+            {
+                serverExclusions += "," + coveragePaths;
+            }
+            else
+            {
+                localExclusions += "," + coveragePaths;
+            }
+            // Remove ServerSettings and LocalSettings property if they exists
+            if (config.ServerSettings.Exists(x => x.Id == SonarExclusions))
+            {
+                config.ServerSettings.RemoveAll(x => x.Id == SonarExclusions);
+            }
+            if (config.LocalSettings.Exists(x => x.Id == SonarExclusions))
+            {
+                config.LocalSettings.RemoveAll(x => x.Id == SonarExclusions);
+            }
+            // Re-add the property with the new value
+            AddSetting(config.ServerSettings, SonarExclusions, serverExclusions);
+            AddSetting(config.LocalSettings, SonarExclusions, localExclusions);
```

Comment:
See my comments on the test files.
You could probably add these only conditionally. There might be something I overlooked here, tests will tell:

```csharp
            // Remove ServerSettings and LocalSettings property if they exists and re-add them with updated values
            if (config.ServerSettings.Exists(x => x.Id == SonarExclusions)
                || !string.IsNullOrWhiteSpace(serverExclusions))
            {
                config.ServerSettings.RemoveAll(x => x.Id == SonarExclusions);
                AddSetting(config.ServerSettings, SonarExclusions, serverExclusions);
            }

            if (config.LocalSettings.Exists(x => x.Id == SonarExclusions)
                || !string.IsNullOrWhiteSpace(localExclusions))
            {
                config.LocalSettings.RemoveAll(x => x.Id == SonarExclusions);
                AddSetting(config.LocalSettings, SonarExclusions, localExclusions);
            }
```

Reply:
What are we trying to fix with this? I added the tests without passing empty params. Can you please clarify what you want to achieve? (if this comment is still relevant)

Reply:
We had a call to clarify. I was also adding the settings when the exclusions were empty. Creating empty local exclusions will always override server exclusions, invalidating them. This was a great catch. It should be fixed now, and I added UTs to test this.